connectivity/tests/health.go

```diff
@@ -45,8 +45,13 @@ func runHealthProbe(ctx context.Context, t *check.ConnectivityTest, pod *check.P
 
 		stdout, err := pod.K8sClient.ExecInPod(ctx, pod.Pod.Namespace, pod.Pod.Name, "", cmd)
 		if err != nil {
-			t.Warnf("cilium-health probe failed: %q, stdout: %q, retrying...", err, stdout)
-			continue
+			select {
+			case <-done:
```

Comment:
this fixes the immediate bug regarding ctx cancellation through Ctrl+C, but as mentioned in https://github.com/cilium/cilium-cli/issues/1314#issuecomment-1376929199 we should also check the retryTimer here.

Reply:
Oh, I will update the pr